dislib/data/classes.py

```diff
@@ -136,9 +141,17 @@ def labels(self):
 
     @property
     def samples(self):
-        self._update_samples()
+        if not self._sparse:
+            self._update_samples()
+        else:
+            self._update_samples_sparse()
+
         return self._samples
 
+    @property
+    def sparse(self):
+        return self._sparse
```

Comment:
it doesn't make sense to have a private attribute _sparse and then expose it as a property called sparse. Just name the attribute sparse and remove the @property

Reply:
It makes sense if you want to make the attribute immutable.